File: pr2 full/2.cpp
#include <iostream>

#include "func.h"

using namespace std;

void runTask2() {
    int c, m;

    cout << "\n--- Задание 2 ---" << endl;
    cout << "Решение сравнения c * d mod m = 1 через u и v" << endl;
    cout << "Введите c: ";
    cin >> c;
    cout << "Введите m: ";
    cin >> m;

    if (m <= 1) {
        cout << "Ошибка: модуль m должен быть больше 1." << endl;
        waitEnter();
        return;
    }

    EuclidResult r = extendedEuclid(m, c);

    cout << "Расширенный алгоритм Евклида:" << endl;
    cout << m << " * " << r.u << " + " << c << " * " << r.v << " = " << r.d << endl;

    if (r.d != 1) {
        cout << "Решения нет, так как НОД(c, m) != 1." << endl;
        waitEnter();
        return;
    }

    int d = positiveMod(r.v, m);
    cout << "Из равенства m*u + c*v = 1 получаем d = v mod m." << endl;
    cout << "d = " << d << endl;
    cout << "Проверка: (c * d) mod m = " << positiveMod(c * d, m) << endl;

    waitEnter();
}

File: pr2 full/Задание 2 Практика Go/2.go
package main

import "fmt"

func runTask2() {
	var c, m int

	fmt.Println("\n--- Задание 2 ---")
	fmt.Println("Решение сравнения c * d mod m = 1 через u и v")
	fmt.Print("Введите c: ")
	fmt.Fscan(reader, &c)
	fmt.Print("Введите m: ")
	fmt.Fscan(reader, &m)
	reader.ReadString('\n')

	if m <= 1 {
		fmt.Println("Ошибка: модуль m должен быть больше 1.")
		waitEnter()
		return
	}

	r := extendedEuclid(m, c)
	fmt.Println("Расширенный алгоритм Евклида:")
	fmt.Printf("%d * %d + %d * %d = %d\n", m, r.U, c, r.V, r.D)

	if r.D != 1 {
		fmt.Println("Решения нет, так как НОД(c, m) != 1.")
		waitEnter()
		return
	}

	d := positiveMod(r.V, m)
	fmt.Println("Из равенства m*u + c*v = 1 получаем d = v mod m.")
	fmt.Println("d =", d)
	fmt.Println("Проверка: (c * d) mod m =", positiveMod(c*d, m))

	waitEnter()
}
